src/telliot_feeds/sources/mimicry/mimicry_mashup_source.py

```diff
@@ -0,0 +1,207 @@
+import asyncio
+from dataclasses import dataclass
+from typing import Any
+from typing import Optional
+
+from aiohttp import ClientConnectionError
+from aiohttp import ClientError
+from aiohttp import ClientResponseError
+from aiohttp import ClientSession
+from aiohttp import ClientTimeout
+from telliot_core.apps.telliot_config import TelliotConfig
+
+from telliot_feeds.datasource import DataSource
+from telliot_feeds.dtypes.datapoint import datetime_now_utc
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+from telliot_feeds.utils.log import get_logger
+
+
+logger = get_logger(__name__)
+
+# for coingecko api chain ids
+block_chains = {"ethereum-mainnet": "ethereum", "solana-mainnet": "solana"}
+
+
+@dataclass
+class NFTMashupSource(DataSource[str]):
+    """
+    DataSource for NFT market-cap mash up expected response data.
+    """
+
+    api_key = TelliotConfig().api_keys.find(name="nftgo")[0].key
+    metric: Optional[str] = None  # ("market_cap")
+    currency: Optional[str] = None  # ("usd")
+    collections: Optional[Any] = None  # ("chain-name", "contract-address")
+    tokens: Optional[Any] = None  # ("chain-name", "token-symbol", "contract-address")
+
+    async def fetch(self, url: str, headers: Optional[dict[str, str]]) -> Optional[Any]:
+        """
+        Fetches data from a given URL.
+
+        Args:
+            url (str): The URL to fetch data from.
+            headers (Optional[dict[str, str]]): Headers to send with the request.
+
+        Returns:
+            Response data from the API, or None if the request fails.
+
+        """
+        async with ClientSession(headers=headers) as session:
+            timeout = ClientTimeout(total=10)
```

Comment:
Ten second wait time rly needed?

Reply:
not sure I understand you correctly, this is so that it doesn't wait for a response indefinitely and that it should timeout after 10 seconds cause thats probably too long or shouldn't take that long. Do you think it should be longer or shorter?